html/semantics/scripting-1/the-script-element/goal-parameter.htm

```diff
@@ -0,0 +1,151 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Goal Parameter on JavaScript MIME</title>
+        <meta http-equiv="content-type" content="text/html; charset=UTF-8" />
+        <meta description="This test checks the Async property on a dynamically-created script element. By default it should be true." />
+        <link rel="author" title="" href="http://www.microsoft.com/" />
+        <link rel="help" href="https://html.spec.whatwg.org/multipage/#dom-script-async"/>
+        <script src="/resources/testharness.js"></script>
+        <script src="/resources/testharnessreport.js"></script>
+    </head>
+    <body>
+      <script type="module">
+
+        function makeTest({
+          fileName,
+          scriptType,
+          contentType,
+          shouldLoad
+        }) {
+          const elem = Object.assign(document.createElement("script"), {
+            type: scriptType,
+            src: `./serve-with-content-type.py?fn=${scriptType === "module" ? "is-module-goal.mjs" : "is-script-goal.js"}&ct=${contentType}`
+          });
+          const name = `${shouldLoad ? "Loads" : "Errors on"} type=${scriptType} when given content-type=${decodeURIComponent(contentType)}`;
+          const t = async_test(name);
+          elem.onload = () => {
+            t.step(() => {
```

Comment:
This will produce better results:
```js
if (!shouldLoad) {
    elem.onload = t.unreached_func('Script should not load');
    elem.onerror = t.step_func_done();
} else {
    elem.onload = t.step_func_done();
    elem.onerror = t.unreached_func("Script should load");
}
```

Reply:
fixed